src/main/java/org/spongepowered/api/data/property/DirectionRelativePropertyHolder.java

```diff
@@ -24,22 +24,50 @@
  */
 package org.spongepowered.api.data.property;
 
-import org.spongepowered.api.data.Property;
 import org.spongepowered.api.util.Direction;
 
 import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
 
-public interface DirectionRelativePropertyHolder {
+public interface DirectionRelativePropertyHolder extends PropertyHolder {
```

Comment:
We can't make them PropertyHolders because they are direction based, it's usually cases of things that literally cannot provide the property by itself